lib/fs.js

```diff
@@ -1617,8 +1617,15 @@ function ReadStream(path, options) {
   if (!(this instanceof ReadStream))
     return new ReadStream(path, options);
 
+  if (options === undefined)
+    options = {};
+  else if (typeof options === 'string')
+    options = { encoding: options };
+  else if (options === null || typeof options !== 'object')
+    throw new TypeError('options must be a string or an object');
+
   // a little bit bigger buffer and water marks by default
-  options = Object.create(options || {});
+  options = Object.create(options);
```

Comment:
Object.create is proposed from https://github.com/nodejs/io.js/pull/635 .
According to the PR, Object.create let `options` inheritable properly.